rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/qos.hpp

```diff
@@ -43,27 +34,98 @@ RMW_FASTRTPS_SHARED_CPP_PUBLIC
 bool
 get_datareader_qos(
   const rmw_qos_profile_t & qos_policies,
-  eprosima::fastrtps::SubscriberAttributes & sattr);
+  eprosima::fastdds::dds::DataReaderQos & dataReaderQos);
 
 RMW_FASTRTPS_SHARED_CPP_PUBLIC
 bool
 get_datawriter_qos(
   const rmw_qos_profile_t & qos_policies,
-  eprosima::fastrtps::PublisherAttributes & pattr);
+  eprosima::fastdds::dds::DataWriterQos & dataWriterQos);
+
+RMW_FASTRTPS_SHARED_CPP_PUBLIC
+bool
+get_topic_qos(
+  const rmw_qos_profile_t & qos_policies,
+  eprosima::fastdds::dds::TopicQos & topicQos);
 
 /*
- * Converts the low-level QOS Policy; of type WriterQos or ReaderQos into rmw_qos_profile_t.
- * Since WriterQos or ReaderQos does not have information about history and depth, these values are not set
+ * Converts the low-level QOS Policy; of type DataWriterQos or DataReaderQos into rmw_qos_profile_t.
```

Comment:
```suggestion
 * Converts the DDS QOS Policy; of type DataWriterQos or DataReaderQos into rmw_qos_profile_t.
```

It is not longer the low-level QOS Policy. Now, DDS QoS are exposed